src/index.ts

```diff
@@ -22,7 +22,7 @@ const createTransformer = (options?: Options) => ({
       : loaders.includes(extName) ? extName: 'text'
     ) as Loader
     const sourcemaps: Partial<TransformOptions> = enableSourcemaps 
-      ? { sourcemap: true, sourcesContent: false, sourcefile: filename } 
+      ? { sourcemap: enableSourcemaps, sourcesContent: false, sourcefile: filename } 
```

Comment:
do we still need this one? sourcemap will be `true` if enableSourcemaps is true, else there is no sourcemap

Reply:
enableSourcemaps is not a flag, but a value. false | true | 'both' | 'inline' | 'external'
That's why I threw it inside there, as some may want a different way of generating them.
I'm more concerned with `opts` and `options` where is passed what? What passed in `options` what passed in `opts`? 
It feels that there is a cleanup needed in the first place.
Plus, well, I guess no one would like to have NO source maps so no debugging possibilities, so I don't know why it's even an option not to generate them, maybe for performance reason only, when built on CI or so, but I'm unsure how source maps affect performance. Since it's an esbuild that compiles stuff I would not bother with that either as it will be still below 1 second.

So amount of options should become smaller with jsx transforms and other useful things, but rest should be default to avoid all these issues I mentioned  above.